src/Connector.php

```diff
@@ -119,12 +119,16 @@ private function listenForPackets(Stream $stream)
         return $deferred->promise();
     }
 
-    private function keepAlive(Stream $stream)
+    private function keepAlive(Stream $stream, $keepAlive)
     {
-        $this->getLoop()->addPeriodicTimer(10, function(Timer $timer) use ($stream) {
-            $packet = new PingRequest($this->version);
-            $this->sendPacketToStream($stream, $packet);
-        });
+        if($keepAlive > 0) {
+            $interval = $keepAlive / 2;
```

Comment:
> It also fixes an issue where the client would get disconnected because it waited right until the Keep Alive time was up before Pinging. After running for a while, I would find that the client would always get disconnected, presumably because at some point it didn't manage to Ping in time.

Before it was only pinging exactly at the time that it might be disconnected. Pinging twice as often gives a lot more leeway, so you aren't risking being disconnected if the server doesn't received your ping in time.